docs/pages/architecture/tls-routing.mdx

```diff
@@ -157,6 +148,73 @@ Teleport proxy's web listener and don't require a local proxy or any special
 ALPN/SNI negotiation. These web connections use standard `http1.1` and `h2`
 protocols for ALPN.
 
+## Working with layer 7 load balancers or reverse proxies (Preview)
+
+Starting from version `13.0`, TLS routing can now be enabled allowing the
+Teleport proxy to serve a single port behind a layer 7 load balancer or reverse
+proxy.
+
+![Layer 7 load balancer setup](../../img/architecture/tls-routing-alb.svg)
+
+It is expected that the layer 7 load balancer or reverse proxy will terminate
+TLS with a public certificate, such as using ACM for AWS ALB. This means that
+the Proxy service does not require a Web TLS certificate using `http_keypair`
+or `acme`.
+
+Teleport clients automatically detect whether Teleport proxy behind a layer 7
+load balancer or a reverse proxy. In such cases, the client initiates a
+connection upgrade similar to "WebSocket" and then sends the TLS routing
+request through the upgraded connection.
+
+![Connection upgrade](../../img/architecture/tls-routing-connection-upgrade.svg)
+
+Non-Teleport clients should require local proxies that can perform the
+special connection upgrades.
+
+Let's take a closer look at how each protocol functions in this configuration.
+
+### SSH
+
+When transporting SSH protocol over TLS routing, `tsh` make connection upgrades
+seamlessly. This is applicable to `tsh ssh/scp` commands, as well as `tsh proxy
+ssh` when connected via `ProxyCommand` using OpenSSH clients.
+
+### Kubernetes
+
+The `tsh proxy kube` command creates a local proxy and an ephemeral kubeconfig
+for Kubernetes clients like `kubectl` to connect to. The local proxy generates
+a self-signed certificate to secure local communications with the Kubernetes
+clients.
+
+When forwarding the requests to the Proxy service, the local proxy performs
+necessary connection upgrades and sets the required SNI for the TLS handshake.
+
+### Databases
+
+In TLS routing mode, `tsh db connect` starts a local proxy which database
+client connection is tunneled through. The local proxy initiates a connection
+to the Proxy service with the connection upgrade then uses database specific
+ALPN values for the TLS handshake.
+
+Similarly, native and GUI clients can connect through `tsh proxy db` which
+starts a local proxy that handles the connection upgrades.
+
+### Web UI and Desktops
+
+The Teleport web UI, along with all its features such as Desktops and AWS
+Console, is fully functional with standard browsers, without any special
+ALPN/SNI values or special connection upgrades.
```

Comment:
```suggestion
The Teleport Web UI is fully functional with standard browsers, without any special
ALPN/SNI values or connection upgrades.
```
I don't think "Desktops" and "AWS Console" are self-evident Web UI features that readers will necessarily grasp, so I think we can remove these mentions.